### ISSUE TITLE
Powerup type argument in plus.txt commands

### ISSUE THREAD

**Issue by kevATin** [biell/alti-server#34]
Could you add the ability to define which types of powerups apply to a zone, key, etc?
This would allow us to easily define different types of powerups in the same powerup spawner as different things. Imagine sometimes getting a portal, sometimes a spawn checkpoint, and sometimes a key, from the same powerup spawner.
Ideally this should allow an array of types in the command.

**Comment by biell** [comment]
I am not sure I understand, can you explain in more detail?  What would be the circumstances which would cause a powerup to provide a different result?  And, how would the player know what they were getting into by choosing that powerup?

**Comment by kevATin** [comment]
/set key A x y shield
/set key B x y wall
/set key C x y missile
/set key D x y bomb

Could all be on the same location, but the powerup gives out different keys based on the randomly chosen powerup type that spawned.
This could be used when you don't want players to be able to get all keys at the same time, because it can make use of the vanilla powerup-cooldown/warmup. Alternatively you could have a script cycle between keys and build in cooldown phases, but that would be much more needlessly complicated I'd say.
You could also do this:
/set key A x y shield
/set key A x y wall
/set key A x y missile
/set key B x y bomb
To only have a 1 in 4 chance of getting key B.
The first three could also be shortened to
/set key A x y shield,wall,missile

As for how the player would know; I'd say the same way the player knows about all keys, portals, etc- either not at all or the map lets them know.

Not sure how do-able this would be for spawnZones; having a key, flag or portal only be available every once in a while should be ok; but having zones pop in and out of existence, would that be likely to cause issues? For example; will a player's zone reset to the start if their current zone gets picked up/the powerup is reset and re-rolled? should they be able to spawn there anyways?
Maybe spawnZones are less of an ideal use for this, but for the other options I think they could be interesting.

**Comment by biell** [comment]
So, you still want keys to perform the same function, you just want one powerup to be able to be more than one key.  This is totally doable, but it won't be easy.

The key's coordinates are its name, and that is how they are looked up.  So, I would have to, for each lookup, probably look up the key by its coordinate pair, and if one didn't exist, then try for the triple of coordinates and type.  So, totally doable, but it will just take time to carefully make sure each time that happens, the if clause is extended to check both.

**Comment by kevATin** [comment]
Yeah I'd like the powerups to have different features/results based on their current powerup type.

If it's too much work then there's no reason to do it now, just keep it in mind as something that could be improved in the future.
The reason why I'd like this feature even though it's already do-able via scripts, is that I like the idea of having as much as possible contained within the map file (plus.txt).
